src/ansible/provisioning.rs

```diff
@@ -0,0 +1,384 @@
+    pub env_variables: Option<Vec<(String, String)>>,
+    pub log_format: Option<LogFormat>,
+    pub logstash_details: Option<(String, Vec<SocketAddr>)>,
+    pub name: String,
+    pub node_count: u16,
+    pub public_rpc: bool,
+}
+
+impl From<DeployOptions> for ProvisionOptions {
+    fn from(deploy_options: DeployOptions) -> Self {
+        ProvisionOptions {
+            beta_encryption_key: deploy_options.beta_encryption_key,
+            binary_option: deploy_options.binary_option,
+            bootstrap_node_count: deploy_options.bootstrap_node_count,
+            env_variables: deploy_options.env_variables,
+            log_format: deploy_options.log_format,
+            logstash_details: deploy_options.logstash_details,
+            name: deploy_options.name,
+            node_count: deploy_options.node_count,
+            public_rpc: deploy_options.public_rpc,
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct AnsibleProvisioner {
+    pub ansible_runner: AnsibleRunner,
+    pub cloud_provider: CloudProvider,
+    pub ssh_client: SshClient,
+}
+
+impl AnsibleProvisioner {
+    pub fn new(
+        ansible_runner: AnsibleRunner,
+        cloud_provider: CloudProvider,
+        ssh_client: SshClient,
+    ) -> Self {
+        Self {
+            ansible_runner,
+            cloud_provider,
+            ssh_client,
+        }
+    }
+
+    pub async fn build_safe_network_binaries(&self, options: &ProvisionOptions) -> Result<()> {
+        let start = Instant::now();
+        println!("Obtaining IP address for build VM...");
+        let build_inventory = self
+            .ansible_runner
+            .get_inventory(AnsibleInventoryType::Build, true)
+            .await?;
+        let build_ip = build_inventory[0].1;
+        self.ssh_client
+            .wait_for_ssh_availability(&build_ip, &self.cloud_provider.get_ssh_user())?;
+
+        println!("Running ansible against build VM...");
+        let extra_vars = self.build_binaries_extra_vars_doc(options)?;
+        self.ansible_runner.run_playbook(
+            AnsiblePlaybook::Build,
+            AnsibleInventoryType::Build,
+            Some(extra_vars),
+        )?;
+        print_duration(start.elapsed());
+        Ok(())
+    }
+
+    pub async fn copy_logs(&self, name: &str, resources_only: bool) -> Result<()> {
+        self.ansible_runner.run_playbook(
+            AnsiblePlaybook::Logs,
+            AnsibleInventoryType::Genesis,
+            Some(format!(
+                "{{ \"env_name\": \"{name}\", \"resources_only\" : \"{resources_only}\" }}"
+            )),
+        )?;
+        self.ansible_runner.run_playbook(
+            AnsiblePlaybook::Logs,
+            AnsibleInventoryType::Nodes,
+            Some(format!(
+                "{{ \"env_name\": \"{name}\", \"resources_only\" : \"{resources_only}\" }}"
+            )),
+        )?;
+        Ok(())
+    }
+
+    pub async fn get_all_node_inventory(&self) -> Result<Vec<(String, IpAddr)>> {
```

Comment:
Should we also get the `bootstrap node's` inventory here?

Reply:
This function just wraps what was in the previous implementation, so it doesn't really have anything to do with this PR, but yeah, I can change it to include all the inventory.